docs/proposals/scheduling/20230415-numa-topology-scheduling.md

```diff
@@ -0,0 +1,238 @@
+            - [Changes in the NodeNUMAResource Plugin](#changes-in-the-nodenumaresource-plugin)
+            - [Changes in the DeviceShare Plugin](#changes-in-the-deviceshare-plugin)
+        - [Changes in Device CRD](#changes-in-device-crd)
+        - [Changes in koordlet](#changes-in-koordlet)
+        - [Compatibility](#compatibility)
+    - [Implementation History](#implementation-history)
+
+<!-- /TOC -->
+
+# Motivation
+
+The scheduler and koordlet of Koordinator have the ability to allocate resources according to NUMA Topology and align them with NUMA Node as much as possible, but currently only support CPU resources. In scenarios such as model training in the field of machine learning, it is required to allocate GPU, RDMA, and CPU at the same time, and also requires them to be on the same NUMA Node or even the same PCI-E, so as to obtain optimal performance.
+
+Although Kubernetes’ kubelet provides Topology Manager, it works on the node side and is not aware of the scheduler. This may cause the scheduled Pod to encounter Topology-related errors when it is scheduled to a certain node, resulting in the Pod being unable to start. And this kind of error is sometimes unacceptable. For example, some job-class workloads may cause the entire job to fail due to this reason. Another problem is that it is impossible to obtain cluster-level optimal NUMA Node resources.
+
+Therefore, a mechanism needs to be implemented to solve this requirement, which can coordinate different resource allocation plugins to uniformly align according to NUMA Topology and obtain optimal performance.
+
+# Goal
+
+1. Provide multiple NUMA Topology management strategies to support flexible resource management requirements.
+1. Provide a mechanism to ensure that when scheduling, CPU/Device and other resources are coordinated and the best/preferred NUMA Node at the Pod-Level is obtained.
+1. The scheduler adds extension points to support the collaborative allocation of different resource allocation plugins.
+1. Describe the capabilities that koordlet needs to guarantee for different NUMA Topology strategies.
+
+# Non-Goals
+
+1. There is no limit to the memory management mechanism under the NUMA architecture. Users can set different kernel parameters according to the actual situation to decide whether to allow memory to be accessed across NUMA Nodes.
+
+# User Stories
+
+## Story 1
+
+The user applies for CPU, GPU and RDMA, and expects the allocated resources to be on the same NUMA node to obtain optimal performance.
+
+## Story 2
+
+The user applies for CPU, GPU and RDMA at the same time, and expects the allocated resources to be aligned according to the NUMA node. At this time, it may be necessary to use multiple NUMA nodes.
+
+# Proposal
+
+## Design Detail
+
+### NUMA Topology Policy
+
+The label `node.koordinator.sh/numa-topology-policy` represents that how to aligning resource allocation according to the NUMA topology. 
```

Comment:
How about considering support policy protocol on pod labels, for example the user story:

There is a workload have performance issue, and I want to make a quick fix to improve it. So I change the label on pod template and redeploy it.

Reply:
It's a good idea. But it also brings additional complexity. The reason is that it is necessary to ensure that Pods with the same policy can be effective together.